libvirt.go

```diff
@@ -300,6 +302,52 @@ func (d *Driver) Create() error {
 	return d.Start()
 }
 
+func (d *Driver) createDisk() error {
+	u, err := url.Parse(d.DiskPathURL)
+	if err != nil {
+		return fmt.Errorf("Invalid URL: %s", d.DiskPathURL)
+	}
+
+	if u.Scheme == "file" || u.Scheme == "" {
+		// For local image on Linux, we can avoid copying by making use of base image through qemu-img
+		// Local disk image so we can avoid copying by making use of backing image feature of qcow2
+		diskPath := u.Path
+		machineIsoPath := filepath.Join(d.StorePath, "machines", d.MachineName, "crc")
+
+		log.Infof("Create disk image at %s (backing store: %s)", machineIsoPath, diskPath)
+		command := exec.Command("qemu-img", "create", "-b", diskPath, "-f", "qcow2", machineIsoPath)
+		output, err := command.CombinedOutput()
+		log.Debugf("Output of `%s` command:\n%s", commandAsString(command), output)
+		if err != nil {
+			log.Warnf("Error occurred running `%s` command: %s", commandAsString(command), err)
+		} else {
+			return nil
+		}
+	}
+
+	b2dutils := mcnutils.NewB2dUtils(d.StorePath, "crc")
+	err = b2dutils.CopyDiskToMachineDir(d.DiskPathURL, d.MachineName)
+
+	return err
+}
+
+// Backport of exec.Cmd.String() for version < 1.13
+func commandAsString(c *exec.Cmd) string {
+	b := new(strings.Builder)
+
+	b.WriteString(c.Path)
+
+	for _, a := range c.Args[1:] {
+
+		b.WriteByte(' ')
+
+		b.WriteString(a)
+
+	}
+
+	return b.String()
+}
```

Comment:
@zeenix why there is so many blank lines here?

Reply:
copy&paste issue from whatever patch web view thing Google uses. :)